pkg/capabilities/consensus/ocr3/aggregators/reduce_test.go

```diff
@@ -0,0 +1,928 @@
+						"FeedID": idBBytes[:],
+					})
+					require.NoError(t, err)
+					return map[commontypes.OracleID][]values.Value{1: {mockValue1}, 2: {mockValue1}, 3: {mockValue2}, 4: {mockValue1}}
+				},
+				shouldReport: true,
+				expectedOutcome: map[string]any{
+					"Reports": []any{
+						map[string]any{
+							"FeedID": idABytes[:],
+						},
+					},
+				},
+				expectedState: map[string]any{
+					"FeedID": idABytes[:],
+				},
+			},
+			{
+				name: "string mode",
+				fields: []aggregators.AggregationField{
+					{
+						InputKey:  "BenchmarkPrice",
+						OutputKey: "Price",
+						Method:    "mode",
+					},
+				},
+				extraConfig: map[string]any{},
+				observationsFactory: func() map[commontypes.OracleID][]values.Value {
+					mockValue1, err := values.WrapMap(map[string]any{
+						"BenchmarkPrice": "1",
+					})
+					require.NoError(t, err)
+					mockValue2, err := values.WrapMap(map[string]any{
+						"BenchmarkPrice": "2",
+					})
+					require.NoError(t, err)
+					return map[commontypes.OracleID][]values.Value{1: {mockValue1}, 2: {mockValue1}, 3: {mockValue2}}
+				},
+				shouldReport: true,
+				expectedOutcome: map[string]any{
+					"Reports": []any{
+						map[string]any{
+							"Price": "1",
+						},
+					},
+				},
+				expectedState: map[string]any{
+					"BenchmarkPrice": "1",
+				},
+			},
+			{
+				name: "bool mode",
+				fields: []aggregators.AggregationField{
+					{
+						InputKey:  "BenchmarkPrice",
+						OutputKey: "Price",
+						Method:    "mode",
+					},
+				},
+				extraConfig: map[string]any{},
+				observationsFactory: func() map[commontypes.OracleID][]values.Value {
+					mockValue1, err := values.WrapMap(map[string]any{
+						"BenchmarkPrice": true,
+					})
+					require.NoError(t, err)
+					mockValue2, err := values.WrapMap(map[string]any{
+						"BenchmarkPrice": false,
+					})
+					require.NoError(t, err)
+					return map[commontypes.OracleID][]values.Value{1: {mockValue1}, 2: {mockValue1}, 3: {mockValue2}}
+				},
+				shouldReport: true,
+				expectedOutcome: map[string]any{
+					"Reports": []any{
+						map[string]any{
+							"Price": true,
+						},
+					},
+				},
+				expectedState: map[string]any{
+					"BenchmarkPrice": true,
+				},
+			},
+			{
+				name: "aggregate on non-indexable type",
+				fields: []aggregators.AggregationField{
+					{
+						// Omitting "InputKey"
+						OutputKey: "Price",
+						Method:    "median",
+					},
+				},
+				extraConfig: map[string]any{},
+				observationsFactory: func() map[commontypes.OracleID][]values.Value {
+					mockValue, err := values.Wrap(1)
+					require.NoError(t, err)
+					return map[commontypes.OracleID][]values.Value{1: {mockValue}, 2: {mockValue}, 3: {mockValue}}
+				},
+				shouldReport: true,
+				expectedOutcome: map[string]any{
+					"Reports": []any{
+						map[string]any{
+							"Price": int64(1),
+						},
+					},
+				},
+				expectedState: map[string]any{"": int64(1)},
```

Comment:
what is `expectedState` vs `expectedOutcome`, as in why doesn't this have `Price` as a key?

Reply:
`expectedState` is how the data is stored within the `metadata`, which is what gets passed in on every new aggregation round as what the "previous round" was.

`expectedOutcome` is what gets put as the workflow step's `result`, which currently is what ends up going to the chain-writer, and then on-chain.